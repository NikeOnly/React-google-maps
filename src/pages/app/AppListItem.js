import React, { Component } from 'react';
import PropTypes from 'prop-types';

class AppListItem extends Component {
    static propTypes = {
        item: PropTypes.shape({}).isRequired,
        onCenterMap: PropTypes.func.isRequired
    }

    onPlaceClick = (e) => {
        e.preventDefault();
        const { item } = this.props;

        this.props.onCenterMap(item.coordinates);
    }

    handleDragStart = (e) => {
        const { item } = this.props;

        e.dataTransfer.setData('place_item', JSON.stringify(item));
    }

    render() {
        const { item } = this.props;

        return (
            <li
                className="app-list-item"
                role="button"
                aria-hidden
                onClick={this.onPlaceClick}

                draggable="true"
                onDragStart={this.handleDragStart}
            >
                <div>{item.name}</div>
                <div>Coordinates: {item.coordinates.lat}, {item.coordinates.lng}</div>
            </li>
        );
    }
}

export default AppListItem;
