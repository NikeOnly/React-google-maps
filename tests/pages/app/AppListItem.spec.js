import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json';
import AppListItem from '../../../src/pages/app/AppListItem';

describe('AppListItem', () => {
    test('match snapshot', () => {
        const container = shallow(
            <AppListItem
                item={{
                    id: 1,
                    name: 'Kyiv',
                    coordinates: { lat: 50.446853, lng: 30.520505 }
                }}
                onCenterMap={jest.fn()}
            />
        );

        expect(toJson(container)).toMatchSnapshot();
    })
})
