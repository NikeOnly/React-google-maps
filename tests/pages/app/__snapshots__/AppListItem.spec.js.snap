// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppListItem match snapshot 1`] = `
<li
  aria-hidden={true}
  className="app-list-item"
  draggable="true"
  onClick={[Function]}
  onDragStart={[Function]}
  role="button"
>
  <div>
    Kyiv
  </div>
  <div>
    Coordinates: 
    50.446853
    , 
    30.520505
  </div>
</li>
`;
